forge-game/src/main/java/forge/game/card/CardProperty.java

```diff
@@ -650,6 +650,10 @@ public static boolean cardHasProperty(Card card, String property, Player sourceC
             if (cards.isEmpty() || !card.equals(cards.get(0))) {
                 return false;
             }
+        } else if (property.startsWith("TopLibrary_")) {
+            CardCollection cards = new CardCollection(card.getOwner().getCardsIn(ZoneType.Library));
```

Comment:
new CardCollection() shouldn't be needed?

Reply:
Ideally all these properties for Top + Bottom would be cleaned up together (with only the Collections.reverse as difference when needed) 🤔
but that could be done later, feel free to resolve